rpc/ws_test.go

```diff
@@ -46,11 +47,12 @@ func (*mockNet) Discover(provides types.ProvidesCoin, searchTime time.Duration)
 	return nil, nil
 }
 func (*mockNet) Query(who peer.AddrInfo) (*net.QueryResponse, error) {
-	return &net.QueryResponse{
-		Offers: []*types.Offer{
-			{ID: testSwapID},
-		},
-	}, nil
+	var offer types.Offer
+	offerJSON := fmt.Sprintf(`{"ID":%q}`, testSwapID.String())
+	if err := json.Unmarshal([]byte(offerJSON), &offer); err != nil {
+		panic(err)
+	}
+	return &net.QueryResponse{Offers: []*types.Offer{&offer}}, nil
```

Comment:
The non-test code doesn't have the need/API to set the offer ID to a specific value, but there was an API to create an offer with a specific ID from JSON, so I just used it here.